<!-- 
  Turtlebot navigation simulation:
  - stage
  - map_server
  - move_base
  - static map
  - amcl
  - rviz view
 -->
<launch>

  <param name="comm_model_filename" value="$(find strategy)/data/comm_model_50.xml" />    <param name="samples_pairs_greedy" value="10000" />
    <param name="samples_leader_multi2" value="100" />
    <param name="samples_follower_multi2_single" value="3" />
    <param name="mindist_vertices_multi2" value="25" /> <param name="mindist_vertices_maxvar" value="25" />
  <arg name="map_file"       default="$(find strategy)/envs/tutorial.yaml"/> 
  <arg name="world_file"     default="$(find strategy)/envs/2_offices.world"/>
  <arg name="robotName1" default="robot_0" />
  <arg name="robotName2" default="robot_1" />
  
  <node pkg="map_server" type="map_server" name="map_server" args="$(arg map_file)" respawn="false" >
    <param name="frame_id" value="/map" />
  </node>

  <node pkg="stage_ros" type="stageros" name="stageros" args="$(arg world_file)" respawn="false">
    <param name="base_watchdog_timeout" value="0.2"/>
  </node>

  <param name="/use_sim_time" value="true"/>

  <!-- robots' common params -->
  <param name="sim" value="1"/>
  <param name="seed" value="3"/>
  <param name="ref_dist" value="1"/>
  <param name="range" value="50" />
  <param name="disc" value="7"/>
  <param name="map_filename" value="$(find strategy)/envs/offices.png"/>
  <param name="polling_signal_period" value="3.5"/> <!--in sec -->   
  <param name="disc_method" value="grid"/>
  <param name="duration" value="1800"/>
  <param name="env_filename" value="$(find strategy)/envs/offices.dat"/>
  <param name="strategy" value="random"/>
  <param name="n_robots" value="2"/>
  <param name="resize_factor" value="0.1"/>
  <param name="tiling" value="0"/>
  <param name="log_folder" value="$(find strategy)/log/"/>

  <!-- BEGIN ROBOT 0 -->
  <group ns="robot_0">
    <param name="tf_prefix" value="$(arg robotName1)" />
    <include file="$(find strategy)/launch/main_robot.launch">
        <arg name="id" value="0" />       
        <arg name="teammates_id" value="1" />
        <arg name="is_leader" value="1" />
    </include>
    <include file="$(find strategy)/includes/single_robot.launch">
    
    	<arg name="namespace" value="$(arg robotName1)"/>
    	<arg name="initial_pose_x" value="33.0"/>
    	<arg name="initial_pose_y" value="17.0"/>
    	<arg name="initial_pose_a" value="0.0"/>
    </include>
  </group>
  <!-- END ROBOT 0 -->


  <!-- BEGIN ROBOT 1 -->
  <group ns="robot_1">
    <param name="tf_prefix" value="$(arg robotName2)" />
    <include file="$(find strategy)/launch/main_robot.launch">
        <arg name="id" value="1" />
        <arg name="teammates_id" value="0" />
        <arg name="is_leader" value="0" />
    </include>
    <include file="$(find strategy)/includes/single_robot.launch">
    	<arg name="namespace" value="$(arg robotName2)"/>
    	<arg name="initial_pose_x" value="33.0"/>
    	<arg name="initial_pose_y" value="15.0"/>
    	<arg name="initial_pose_a" value="0.0"/>
    </include>
  </group>
  <!-- END ROBOT 1 -->


</launch>
